apps/server/src/modules/authentication/testing/icurrentuser.factory.ts

```diff
@@ -0,0 +1,56 @@
+import { BaseFactory } from '@shared/testing';
+import { ObjectId } from 'bson';
+import { ICurrentUser } from '../interface';
+
+class CurrentUser implements ICurrentUser {
+	userId: string;
+
+	roles: string[];
+
+	schoolId: string;
+
+	accountId: string;
+
+	systemId: string;
+
+	isExternalUser: boolean;
+
+	constructor(data: ICurrentUser) {
+		this.userId = data.userId;
+		this.roles = data.roles;
+		this.schoolId = data.schoolId;
+		this.accountId = data.accountId;
+		this.systemId = data.systemId || '';
+		this.isExternalUser = data.isExternalUser;
+	}
+}
+
+export class ICurrentUserFactory extends BaseFactory<CurrentUser, ICurrentUser> {

```

Comment:
without I at the begin. I think I is just showing that currentUser an interface is.
```suggestion
export class CurrentUserFactory extends BaseFactory<CurrentUser, ICurrentUser> {
```